doc/dev/debugging.rst

```diff
@@ -324,42 +324,27 @@ runtime libraries. The generated shared object is stored by the caching mechanis
 for future calls.
 
 Recompiling a Function
-=================
+======================
+
 Catalyst offers a way to extract IRs from pipeline stages and feed modified IRs back for recompilation.
 To enable this feature, ``qjit`` decorated function must be compiled with the option ``keep_intermediate=True``.
 
 The following example creates a square function decorated with ``@qjit(keep_intermediate=True)``.
 The function must be compiled first so that the IR from each pipeline stage can be accessed.
 
-.. code-block:: python
-
-    @qjit(keep_intermediate=True)
-    def f(x):
-        return x**2
-    f(2.0)
-    >> 4.0
+>>> @qjit(keep_intermediate=True)
+... def f(x):
+...     return x**2
+>>> f(2.0)
+4.0
 
-After compilation, we can use :func:`~.debug.get_compilation_stage`  in the ``debug`` module to get the IR from the given compiler stage.
-:func:`~.debug.get_compilation_stage` accepts a ``qjit`` decorated function and a stage name in string. It return the IR after the
-given stage.
+After compilation, we can use :func:`~.debug.get_compilation_stage`  in the ``debug`` module to get the IR
+from the given compiler stage. In this example, we request for the IR after ``HLOLoweringPass``:
 
-The available options are:
+>>> from catalyst.debug import get_compilation_stage
+>>> old_ir = get_compilation_stage(f, "HLOLoweringPass")
 
-* MLIR stages: ``mlir``, ``HLOLoweringPass``, ``QuantumCompilationPass``, ``BufferizationPass`` and ``MLIRToLLVMDialect``.
-* LLVM stages: ``llvm_ir``, ``CoroOpt``, ``O2Opt``, ``Enzyme``, and ``last``.
-
-Note that compiled functions might not always have ``CoroOpt``, ``O2Opt``, and ``Enzyme`` stages.
-The option ``last`` will provide the IR right before generating its object file.
```

Comment:
I removed this here, since it is mostly a copy and paste from the docstring. best to have readers go to the docstring for the fine details, as it is likely we will forget to maintain this and it will go out of sync